addons/account/models/account_move_line.py

```diff
@@ -2903,6 +2924,24 @@ def remove_move_reconcile(self):
         """ Undo a reconciliation """
         (self.matched_debit_ids + self.matched_credit_ids).unlink()
 
+    def _reconcile_marked(self):
+        temp_numbers = list({
+            line.matching_number
+            for line in self
+            if line.matching_number and line.matching_number.startswith('I')
+        })
+        if temp_numbers:
+            for _matching_number, account, lines in self._read_group(
+                domain=[('matching_number', 'in', temp_numbers)],
+                groupby=['matching_number', 'account_id'],
+                aggregates=['id:recordset'],
+            ):
+                if all(move.state == 'posted' for move in lines.move_id):
```

Comment:
could group by parent_state above, to avoid looping on each move

Reply:
More complex code and we will most likely eventually loop on the values anyway :shrug: 
Really worth?

Reply:
no entirely sure myself. Let's consider my comment as premature optimization :) it's fine like that